lib/internal/mime.js

```diff
@@ -0,0 +1,279 @@
+    let parameterName = exec(/^[^;=]*/, StringPrototypeSlice(str, position));
+    parameterName = parameterName[0];
+    position += parameterName.length;
+    parameterName = toASCIILower(parameterName);
+    if (position < str.length) {
+      const [codePoint] = StringPrototypeSlice(str, position);
+      if (codePoint === ';') {
+        continue;
+      } else {
+        position += 1;
+      }
+    }
+    if (position >= str.length) {
+      break;
+    }
+    const [codePoint] = StringPrototypeSlice(str, position);
+    let parameterValue = null;
+    if (codePoint === '"') {
+      position += 1;
+      const inside = exec(/^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u, StringPrototypeSlice(str, position));
+      position += inside[0].length;
+      // Trailing \ at end of string
+      parameterValue = StringPrototypeReplace(StringPrototypeSlice(
+        inside[0],
+        0,
+        inside[1] || inside[2] ? -1 : undefined
+      ), /[\\]([\s\S])/ug, '$1');
+      if (inside[1]) parameterValue += '\\';
+    } else {
+      let value = exec(/[^;]*/, StringPrototypeSlice(str, position))[0];
+      position += value.length;
+      value = StringPrototypeReplace(value, ENDING_WHITESPACE, '');
+      if (value === '') continue;
+      parameterValue = value;
+    }
+    if (parameterName !== '' &&
+      test(NotHTTPTokenCodePoint, parameterName) === false &&
+      test(NotHTTPQuotedStringCodePoint, parameterValue) === false &&
+      parameterName in parameters === false) {
+      parameters[parameterName] = parameterValue;
+    }
+  }
+  target.parameters = ObjectEntries(parameters);
+  return target;
+}
+
+const test = (pattern, value) => {
+  pattern.lastIndex = 0;
+  const ret = pattern.test(value);
+  // console.dir({
+  //   pattern,
+  //   value,
+  //   _: [...value].map((c) => c.codePointAt(0)),
+  //   ret,
+  //   _1: (pattern.lastIndex = 0, NotHTTPTokenCodePoint.exec(value))
+  // });
+  return ret;
+};
+const exec = (pattern, value) => {
+  pattern.lastIndex = 0;
+  const ret = pattern.exec(value);
+  // console.dir({
+  //   pattern,
+  //   value,
+  //   _: [...value].map((c) => c.codePointAt(0)),
+  //   ret,
+  //   _1: (pattern.lastIndex = 0, NotHTTPTokenCodePoint.exec(value))
+  // });
+  return ret;
+};
+
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  NotHTTPTokenCodePoint.lastIndex = 0;
+  const encode = test(NotHTTPTokenCodePoint, value);
+  if (!encode) return value;
+  const escaped = StringPrototypeReplace(value, /["\\]/g, '\\$&');
+  const ret = `"${escaped}"`;
+  return ret;
+};
+
+const MIMEStringify = (type, subtype, parameters) => {
+  let ret = `${type}/${subtype}`;
+  const entries = MIMEParamsData(parameters).entries();
+  let keyValuePair, done;
+  for (;;) {
+    ({ value: keyValuePair, done } =
+        MapIteratorPrototypeNext(entries));
+    if (done) break;
+    const [key, value] = keyValuePair;
+    const encoded = encode(value);
+    ret += `;${key}=${encoded}`;
+  }
```

Comment:
Why not a simple for of loop?

Reply:
entries() returns an iterator for maps, and that would mean i need to handle if someone overwrites [Symbole.iterator] on this, this would still work even if someone does funky things to the iterator via prototype stuff.

Reply:
That seems a bit overkill in my opinion. You mean that this would work if someone override `[Symbol.iterator]` for `Map`?. If someone does that it already potentially breaks in so many other places that I would rather prioritise readability.

Reply:
I will not change this code as it one of its intent is in potentially being used when loading things, I could extract this into a primordial helper for readability if desired and cover it with comments but would not see readability as the key for things in the loader pipeline until policy assertions run. After policy assertions the [general idea is to let the users do whatever](https://docs.google.com/document/d/1h__FmXsEWRuNrzAV_l3Iw9i_z8fCXSokGfBiW8-nDNg/edit)

Reply:
I don't understand most of that so you probably know better :).